### ISSUE TITLE
[Bug]: Reading additional metadata from SharePoint using GetFolderFilesByServerRelativeUrl

### ISSUE THREAD

**Issue by dorianwoolger** [microsoft/BCApps#499]
### Describe the issue

I am trying to use the SharePoint module to read additional metadata on the items.
I see there is the following IntegrationEvent which refers to exactly what I want to do.
```
    [IntegrationEvent(false, false)]
    /// <summary>
    /// Process SharePointFile Metadata - Use to extract custom meta data into model record
    /// </summary>
    /// <remarks>Extend the "SharePoint File" table to store any custom data.</remarks>
    /// <param name="Metadata">__metadata node of SharePointFile Json Object</param>
    /// <param name="SharePointFile">SharePointFile temporary record.</param>
    internal procedure ProcessSharePointFileMetadata(Metadata: JsonToken; var SharePointFile: Record "SharePoint File" temporary)
    begin
    end;
```

It refers to extending the "SharePoint File" table but the table is not extendable. I could work round that but still think it needs fixing.

The bigger issues is the JsonToken that is passed is just the file Id and no metadata.
The below is take from codeunit 9106 "SharePoint File"
```
        if Payload.Get('ListItemAllFields', JToken) then begin
            Payload := JToken.AsObject();

            if Payload.Get('Id', JToken) then
                SharePointFile.Id := JToken.AsValue().AsInteger();

            SharePointClient.ProcessSharePointFileMetadata(JToken, SharePointFile);
        end;
```

Should the the Payload JsonObject not be the thing that is passed?

### Expected behavior

1. Ability to extend the "SharePoint File" table
2. To be passed all metadata to the integration event

### Steps to reproduce

1. 
![image](https://github.com/microsoft/BCApps/assets/19670620/089a9fd6-4926-4c96-9cad-93d0f99e88b3)

2. I would expect the below to show Json of the metadata

```
    [EventSubscriber(ObjectType::Codeunit, Codeunit::"SharePoint Client", 'ProcessSharePointFileMetadata', '', false, false)]
    local procedure ProcessSharePointFileMetadata(Metadata: JsonToken; var SharePointFile: Record "SharePoint File" temporary)
    begin
        Message(format(Metadata));
    end;
```

### Additional context

n/a

**Comment by JesperSchulz** [comment]
I'm not super strong in how the SharePoint module works (yet). We would have to rely on community members which have tried using it. Maybe create a draft PR to show your intended changes and see if anyone can assist?

@dynaway-kmi or @mynavblog, would you be able to assess if this change is good?

**Comment by dorianwoolger** [comment]
@JesperSchulz, thanks for the reply, will get the changes drafted up over the next few days.

**Comment by dynaway-kmi** [comment]
Hi,

I have run into the same problems when trying to use metadata within our project.

I have already prepared a fix:
https://github.com/dynaway-kmi/BCApps/commit/cab44c0628059c4816a032ff8366b1455e8e1f3b
Should I make a PR out of it?

**Comment by JesperSchulz** [comment]
> Hi,
> 
> I have run into the same problems when trying to use metadata within our project.
> 
> I have already prepared a fix: [dynaway-kmi@cab44c0](https://github.com/dynaway-kmi/BCApps/commit/cab44c0628059c4816a032ff8366b1455e8e1f3b) Should I make a PR out of it?

Yes please! You can link to this issue.
